File: ce/snippets/visualbasic/CRMV8/queryvb/vb/linqexamples13.vb
' <snippetlinqexamples13>


 Using orgSvcContext As New OrganizationServiceContext(_serviceProxy)
  Dim query_join9 = From a In orgSvcContext.CreateQuery("account") _
                    Group Join c In orgSvcContext.CreateQuery("contact") _
                    On a("primarycontactid") Equals c("contactid") Into gr = _
                    Group From c_joined In gr.DefaultIfEmpty() _
                    Select New With {Key .account_name = a.Attributes("name")}
  For Each c In query_join9
   Console.WriteLine(c.account_name)
  Next c
 End Using

' </snippetlinqexamples13>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/linqexamples13.cs
// <snippetlinqexamples13>


 using (OrganizationServiceContext orgSvcContext = new OrganizationServiceContext(_serviceProxy))
 {
  var query_join9 = from a in orgSvcContext.CreateQuery("account")
                    join c in orgSvcContext.CreateQuery("contact") 
                    on a["primarycontactid"] equals c["contactid"] into gr
                    from c_joined in gr.DefaultIfEmpty()
                    select new
                    {
                     account_name = a.Attributes["name"]
                    };
  foreach (var c in query_join9)
  {
   System.Console.WriteLine(c.account_name);
  }
 }

// </snippetlinqexamples13>